### PR TITLE
Persian translation

### PR DESCRIPTION
**Pull request by mnameghi**
all strings translated and checked again.
run `tools/unix/generate_localizations.sh` tool and generate `fa` translation for android and iphone.

### CODE REVIEW

**Comment by mapsmetest** [comment]
Can one of the admins verify this patch?

**Comment by herotux** [comment]
my good freind @mnameghi have a better pull request please merge that:

**Comment by herotux** [comment]
good work @mnameghi i hope accept this soon

**Comment by imanmoghimiq30** [comment]
That's good
God bless you =))

**Comment by bykoianko** [comment]
Thank you @mnameghi

File: data/strings/strings.txt [hunk 1]
```diff
@@ -24215,6 +24241,7 @@ fa = دوباره نپرس
     zh-Hant = • 更精準估計到達時間。\n• 該應用程式已經考慮到限制性區域和道路標誌。
     el = • Ακριβέστερη εκτίμηση ώρας άφιξης. \n• Η εφαρμογή λαμβάνει υπόψη τις περιοχές που απαγορεύονται και την οδική σήμανση.
     sk = • Presnejší odhad času príchodu.\\n • Aplikácia berie do úvahy obmedzenia a dopravné značky.
+    fa = تخمین دقیق تر از زمان ورود. \ n • برنامه به مناطق محدود و نشانه های جاده اشاره دارد.
```
Comment (mpimenov) [inline data/strings/strings.txt]: "\n" should not contain spaces in between

File: data/strings/strings.txt [hunk 2]
```diff
@@ -24335,6 +24365,7 @@ fa = دوباره نپرس
     zh-Hant = 點按地點，即可向您的路線新增路徑點。
     el = Πατήστε στα μέρη για να προσθέσετε ενδιάμεσα σημεία στη διαδρομή σας.
     sk = Ťuknite na miesta, ak chcete do trasy pridať body.
+    fa = مکان ها را برای افزودن ایستگاه های راه به مسیر خود ضربه بزنید
```
Comment (mpimenov) [inline data/strings/strings.txt]: Dot in the end, please.

File: data/strings/strings.txt [hunk 3]
```diff
@@ -24607,6 +24646,7 @@ fa = دوباره نپرس
     zh-Hant = 無法載入資訊。
     el = Η φόρτωση πληροφοριών απέτυχε.
     sk = Nepodarilo sa načítať informácie.
+    fa = اطلاعات بارگیری نشد
```
Comment (mpimenov) [inline data/strings/strings.txt]: Dot.

File: data/strings/strings.txt [hunk 4]
```diff
@@ -25171,6 +25227,7 @@ fa = دوباره نپرس
     el = Γενικό σχόλιο
     sk = Všeobecné pripomienky
     sw = Maoni Jumla
+    fa =  "تنظیمات: دکمه "ارسال بازخورد کلی"
```
Comment (mpimenov) [inline data/strings/strings.txt]: I don't think the quotes are correct here.

File: data/strings/strings.txt [hunk 5]
```diff
@@ -25295,7 +25355,8 @@ fa = دوباره نپرس
...
     sw = Kwa baadhi ya lugha, utahitaji kusakinisha programu nyingine ya kusanidi usemi au kifurushi cha ziada cha lugha kutoka kwenye duka la programu-tumizi (Google Play Market, Samsung Apps).\nFungua mipangilio ya kifaa chako → Lugha na Ingizo → Usemi → Tokeo la maandishi kuwa usemi.\nHapa unaweza kusimamia usanidi wa usemi (kwa mfano, kupakua kifurushi cha lugha ili utumie nje ya mtandao) na uchague injini nyingine ya maandishi-kuwa-usemi.
-
+    fa = برای بعضی از زبان ها، شما باید speech synthesizer یا یک بسته زبان دیگر را از فروشگاه برنامه (بازار Google Play، Samsung Apps) نصب کنید. \ n تنظیمات دستگاه خود را باز کنید. → زبان و ورودی → گفتار → خروجی متن به گفتار. می توانید تنظیمات برای ترکیب گفتار (به عنوان مثال، بسته های زبان را برای استفاده آفلاین دانلود کنید) را مدیریت کنید و یک موتور دیگر متن به گفتار را انتخاب کنید.
```
Comment (mpimenov) [inline data/strings/strings.txt]: \n

File: data/strings/strings.txt [hunk 6]
```diff
@@ -26952,7 +27063,7 @@ fa = دوباره نپرس
     el = Σύμφωνα με %d κριτικές
     sk = Na základe %d recenzií
     sw = Kulingana na maoni %d
-
+    fa = براساس %d ارزیابی
```
Comment (mpimenov) [inline data/strings/strings.txt]: Please leave the empty line before the next block.

File: data/strings/strings.txt [hunk 7]
```diff
@@ -26983,6 +27094,7 @@ fa = دوباره نپرس
     el = Πατήστε για να βαθμολογήσετε και να γράψετε μια κριτική για αυτό το μέρος.
     sk = Ťuknutím ohodnoťte a preskúmajte toto miesto.
     sw = Gusa ili kukadiria na kutoa maoni kuhusu mahali hapa.
+    fa = برای رای دادن و ارزیابی این مکان ضربه بزنید
```
Comment (mpimenov) [inline data/strings/strings.txt]: Dot.

File: data/strings/strings.txt [hunk 8]
```diff
@@ -28619,6 +28779,7 @@ fa = دوباره نپرس
     nb = Ikke tilgjengelig offline.
     sk = Nie sú k dispozícii offline.
     sw = Haipatikani nje ya mtandao.
+    fa = در دسترس نیست
```
Comment (mpimenov) [inline data/strings/strings.txt]: Dot.

File: data/strings/strings.txt [hunk 9]
```diff
@@ -29439,6 +29623,9 @@ fa = دوباره نپرس
     sw = Alamisho %d
     zh-Hans = %d 個書籤
     zh-Hant = %d個書籤
+    fa:one = %d نشانه
+    af:other = %d نشانه‌ها
```
Comment (mpimenov) [inline data/strings/strings.txt]: s/af/fa/

File: data/strings/strings.txt [hunk 10]
```diff
@@ -29439,6 +29623,9 @@ fa = دوباره نپرس
...
+
```
Comment (mpimenov) [inline data/strings/strings.txt]: One empty line is enough. Remove this one, please.

File: data/strings/strings.txt [hunk 11]
```diff
@@ -29604,6 +29795,7 @@ fa = دوباره نپرس
     sw = Inatekeleza %s...
     zh-Hans = 正在套用 %s...
     zh-Hant = 應用 %s 中……
+    fa = اعمال
```
Comment (mpimenov) [inline data/strings/strings.txt]: "..."

File: data/strings/strings.txt [hunk 12]
```diff
@@ -33648,7 +33971,7 @@ fa = دوباره نپرس
     da = Vil du finde populære steder i nærheden?
     nl = Wilt u populaire plaatsen in de omgeving vinden?
     fi = Haluatko löytää suosittuja lähistöllä olevia kohteita
-    fr = Vous voulez trouver des endroits populaires à proximité ?
+    fr = Vous voulez trouver des endroits populaires à proximité ?
```
Comment (mpimenov) [inline data/strings/strings.txt]: What happened here? Please revert.

File: data/strings/strings.txt [hunk 13]
```diff
@@ -34118,7 +34456,7 @@ fa = دوباره نپرس
     ja = インターネットに接続して、ホテルを日付で絞り込み検索します。
     ko = 인터넷에 연결해 날짜를 통해서 호텔들을 필터 하세요.
     no = Koble seg til Internett for å velge ankomst- og avreisedato.
-    pl = Połącz się z Internetem aby wybrać datę zameldowania i wymeldowania.
+    pl = Połącz się z Internetem aby wybrać datę zameldowania i wymeldowania.
```
Comment (mpimenov) [inline data/strings/strings.txt]: Please revert.

File: data/strings/strings.txt [hunk 14]
```diff
@@ -31620,11 +31872,12 @@ fa = دوباره نپرس
     sw = Orodha zilizopakuliwa
     zh-Hans = 上传的列表
     zh-Hant = 加載的列表
+    fa = لیست‌های دانلودشده
 
   [author_name_by_prefix]
     en = by %s
     ru = %s
-    ar = بواسطة%s
+    ar = بواسطة٪ s
```
[the remark this answers is not in the document]
Comment (mnameghi) [reply data/strings/strings.txt]: ٪ is wrong! It must be %

File: data/strings/strings.txt (hunk 13 above)
Comment (herotux) [reply data/strings/strings.txt]: Nothing just lines changed after translate and add new lines

File: data/strings/strings.txt [hunk 15]
```diff
@@ -33408,7 +33721,8 @@ fa = دوباره نپرس
     sw = %s/mwezi (okoa %s/mwaka)
     zh-Hans = %s/月 (节省％s /年)
     zh-Hant = %s/月 (節省％s /年)
-
+    fa = ماه (صرفه جویی %s/ سال)
```
[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: Yes its ok

File: data/strings/strings.txt [hunk 16]
```diff
@@ -33441,6 +33755,7 @@ fa = دوباره نپرس
     sw = %s/wiki (hakuna kuokoa)
     zh-Hans = %s/周 (未节省)
     zh-Hant = %s/周 (未節省)
+    fa = %s/یک هفته (بدون صرفه جویی)
```
[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: Yes this is not problem

File: data/strings/strings.txt (hunk 12 above)
Comment (herotux) [reply data/strings/strings.txt]: No problem just lines changed after add new string in fa

File: data/strings/strings.txt [hunk 17]
```diff
@@ -32566,6 +32850,7 @@ fa = دوباره نپرس
     sw = …zaidi
     zh-Hans = …更多
     zh-Hant = …更多
+    fa = بیشتر...
```
[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: This is because github in app shown right

File: data/strings/strings.txt (hunk 12 above)
Comment (herotux) [reply data/strings/strings.txt]: nothing changed i checked again its right

File: data/strings/strings.txt (hunk 13 above)
Comment (herotux) [reply data/strings/strings.txt]: nothing changed i checked again its right

File: data/strings/strings.txt [hunk 18]
```diff
@@ -30006,6 +30209,7 @@ fa = دوباره نپرس
     sw = Jina halikuweza kuwa tupu
     zh-Hans = 名字不能为空
     zh-Hant = 名字不能為空
+    fa = نام ممکن نیست خالی باشد
```
[the remark this answers is not in the document]
Comment (mpimenov) [reply data/strings/strings.txt]: This string should not be translated at all.

File: data/strings/strings.txt [hunk 19]
```diff
@@ -32104,6 +32374,7 @@ fa = دوباره نپرس
     sw = Kifaa chako cha mkononi kinaweza kutoa mpangilio wa “Zuia Ufuatiliaji wa Matangazo” au “Chagua kutoka kwenye utangazaji unaotegemea mapendeleo”.\n\nKwa Huduma za Android na Google Play toleo 4.0 na zaidi:\nMipangilio → Google → Matangazo → Chagua kutoka kwa Ubinafsishaji wa Matangazo\nau\nMipangilio → Google → Akaunti ya Google → Maelezo ya kibinafsi na faragha → Mipangilio ya Matangazo → Ubinafsishaji wa Matangazo
     zh-Hans = 您可以选择不在设备设置中接收定向广告。\ NAndroid和Google Play Services 4.0或更高版本\ n \ n电话设置→Google→广告→停用广告个性化\ n或\ n手机设置→Google→Google帐户→隐私→首选项设置→ 广告的个性化
     zh-Hant = 您可以選擇停止在設備設置中接受定向廣告。Android и Google Play Services 4.0或更高版本\n\n電話設置 → Google → 廣告 → 停用廣告個性化\n或\n電話設置 → Google → Google賬戶 → 隱私 → 首選項設置 → 廣告個性化
+    fa = دستگاه تلفن همراه شما ممکن است یک «ردیابی محدود آگهی» یا «لغو تبلیغات مبتنی بر علاقه» ارائه کند. \ n \ n برای Android و Google Play Services نسخه 4.0 و بالاتر: \ n تنظیمات → Google → آگهی → غیرفعال کردن تبلیغات شخصی سازی \ یا \ n تنظیمات → Google → حساب Google → اطلاعات شخصی و حریم خصوصی → تنظیمات تبلیغات → شخصی سازی تبلیغات
```
[the remark this answers is not in the document]
Comment (mpimenov) [reply data/strings/strings.txt]: I think there's one slash left over that should be removed.

File: data/strings/strings.txt (hunk 12 above)
Comment (mpimenov) [reply data/strings/strings.txt]: No, it has changed.
The old character was NO-BREAK SPACE (hex codepoint 00A0).
The new character is SPACE (hex codepoint 0020).

The old one was correct. Please revert.
https://en.wikipedia.org/wiki/Non-breaking_space

File: data/strings/strings.txt (hunk 18 above)
[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: why?

[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: ok i will remove theme

File: data/strings/strings.txt (hunk 11 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 10 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 9 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 8 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 7 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 6 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 5 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 2 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 4 above)
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 19 above)
[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt [hunk 20]
```diff
@@ -32764,6 +33054,7 @@ fa = دوباره نپرس
     sw = Sehemu ya "Gundua yaliyo karibu" itakusaidia kupata maeneo maarufu zaidi jijini.
     zh-Hans = “探索周围”部分将帮助您找到该市最受欢迎的地方。
     zh-Hant = “探索鄰近區域”部分將幫您找到該城市最受歡迎的地方。
+    fa =  بخش کشف نزدیک به شما کمک خواهد کرد که مکان های محبوب شهر را پیدا کنید.
```
[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: fixed

File: data/strings/strings.txt (hunk 12 above)
Comment (herotux) [reply data/strings/strings.txt]: yes i found it 
 fixed

File: data/strings/strings.txt (hunk 18 above)
[the remark this answers is not in the document]
Comment (mpimenov) [reply data/strings/strings.txt]: Oh wait. It's ok, either I missed something or it is another github's bug due to the size of the file.

[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: ok :rose:

File: data/strings/strings.txt [hunk 21]
```diff
@@ -33639,6 +33961,7 @@ fa = دوباره نپرس
     sw = Pakua na upange safari za watarii wanaozuru sehemu za kuvutia katika miji mbali mbali ya ndunia.
     zh-Hans = 下载并设置到世界各地各个城市景点的旅游路线。
     zh-Hant = 下載并設置到世界各地各個城市景點的旅遊路線。
+    fa = دانلود و برنامه ریزی مسیرهای توریستی در مکان های جالب در شهرهای مختلف جهان  .
```
[the remark this answers is not in the document]
Comment (herotux) [reply data/strings/strings.txt]: fixed

**Comment by herotux** [comment]
we fixed all conflicts please  review again

**Comment by herotux** [comment]
just some lines are moved a line.
no remove no edit just persian strings added.

**Comment by mpimenov** [comment]
Please sign the CLA: https://maps.me/cla/

File: data/strings/strings.txt [hunk 22]
```diff
@@ -23166,6 +23166,7 @@ fa = دوباره نپرس
     zh-Hant = 4,000 家餐廳的快速外送。向 Delivery Club 訂餐。
     el = Γρήγορη παράδοση από 4.000 εστιατόρια. Παραγγείλετε από το Delivery Club.
     sk = Rýchle dodanie zo 4 000 reštaurácií. Objednávky od spoločnosti Delivery Club.
+    fa = تحویل سریع از 4000 رستوران. سفارش از تحویل باشگاه.
```
Comment (mpimenov) [inline data/strings/strings.txt]: 4،000 ?

File: data/strings/strings.txt [hunk 23]
```diff
@@ -24275,6 +24303,7 @@ fa = دوباره نپرس
     zh-Hant = Maps 所需存儲空間減少 20%。
     el = Οι χάρτες απαιτούν 20% λιγότερο αποθηκευτικό χώρο.
     sk = Mapy Maps vyžadujú až o 20% menej ukladacieho priestoru.
+    fa = نقشه ها نیاز به ذخیره سازی 20٪ کمتر دارند.
```
Comment (mpimenov) [inline data/strings/strings.txt]: Wrong %

Comment (mpimenov) [reply data/strings/strings.txt]: Oh, this one is not wrong in fact.

File: data/strings/strings.txt (hunk 19 above)
[the remark this answers is not in the document]
Comment (mpimenov) [reply data/strings/strings.txt]: Is it though?
I mean here
```
Google → آگهی → غیرفعال کردن تبلیغات شخصی سازی \ یا
```

File: data/strings/strings.txt (hunk 21 above)
[the remark this answers is not in the document]
Comment (mpimenov) [reply data/strings/strings.txt]: Did you push the changes?

File: data/strings/strings.txt (hunk 12 above)
Comment (mpimenov) [reply data/strings/strings.txt]: Did you push the changes?

**Comment by mpimenov** [comment]
Please rebase the PR to the master branch.

**Comment by mnameghi** [comment]
Signed

**Comment by mpimenov** [comment]
Please rebase onto mapsme:master and we'll merge.

**Comment by mpimenov** [comment]
I think the two last comments should be squashed into one.

**Comment by mnameghi** [comment]
How can I do it?

**Comment by mpimenov** [comment]
git rebase -i

**Comment by mpimenov** [comment]
See the message at the bottom: "This branch cannot be rebased due to conflicts".
This is because it includes a merge commit, the one with the unresolved conflict.
Please perform an interactive rebase, resolve the conflicts and force push the result here again. Until then github won't allow us to merge it.

This link may be useful: https://stackoverflow.com/questions/10697463/resolve-git-merge-conflicts-in-favor-of-their-changes-during-a-pull

**Comment by mpimenov** [comment]
If git doesn't show errors when you type locally "git rebase upstream master" using the latest version of our master branch, it's a good sign that everything is safe and will work here too.

**Comment by herotux** [comment]
bcause of my problem in rebase and understand problem that you noticed me i am create new pull request from lastest edits in master you can see theme here please merge that thanks .

**Comment by herotux** [comment]
#9627

File: data/strings/strings.txt [hunk 24]
```diff
@@ -26000,6 +26081,7 @@ fa = دوباره نپرس
     el = Κάρτα μνήμης
     sk = Pamäť na karte
     sw = Kumbukumbu ya kadi
+    sw = کارت حافظه
```
Comment (tatiana-yan) [inline data/strings/strings.txt]: fa = ?

**Comment by tatiana-yan** [comment]
@mnameghi there is a problem with new PR: it's only autogenerated files in PR (no strings.txt which is source for autogeneration). So after automatic generation all fa translations will disappear.
I created PR with your translations, fixed sw/fa, rebased it and added autogenerated translations in separate commit: https://github.com/mapsme/omim/pull/9630 
I hope it'll be easier way. Is it ok? Please review it.

**Comment by mpimenov** [comment]
Thank you. We'll merge the changes in #9630.

If you are interested, synonyms that we use for the search are stored in data/categories.txt and data/categories_cuisines.txt and probably need proofreading too.
https://github.com/mapsme/omim/blob/master/data/categories.txt
https://github.com/mapsme/omim/blob/master/data/categories_cuisines.txt

**Pull request closed without merging**